packages/driver/src/cy/commands/sessions/manager.ts

```diff
@@ -0,0 +1,295 @@
+import _ from 'lodash'
+import { $Location } from '../../../cypress/location'
+
+import {
+  getCurrentOriginStorage,
+  setPostMessageLocalStorage,
+  getPostMessageLocalStorage,
+} from './utils'
+
+type ActiveSessions = Cypress.Commands.Session.ActiveSessions
+type SessionData = Cypress.Commands.Session.SessionData
+
+export default class SessionsManager {
+  Cypress
+  cy
+  currentTestRegisteredSessions = new Map()
+
+  constructor (Cypress, cy) {
+    this.Cypress = Cypress
+    this.cy = cy
+  }
+
+  setActiveSession = (obj: ActiveSessions) => {
+    const currentSessions = this.cy.state('activeSessions') || {}
+
+    const newSessions = { ...currentSessions, ...obj }
+
+    this.cy.state('activeSessions', newSessions)
+  }
+
+  getActiveSession = (id: string): SessionData => {
+    const currentSessions = this.cy.state('activeSessions') || {}
+
+    return currentSessions[id]
+  }
+
+  clearActiveSessions = () => {
+    const curSessions = this.cy.state('activeSessions') || {}
+    const clearedSessions: ActiveSessions = _.mapValues(curSessions, (v) => ({ ...v, hydrated: false }))
+
+    this.cy.state('activeSessions', clearedSessions)
+  }
+
+  mapOrigins = async (origins: string | Array<string>): Promise<Array<string>> => {
+    const getOrigins = this.Cypress.Promise.map(
+      ([] as string[]).concat(origins), async (v) => {
+        if (v === '*') {
+          return await this.getAllHtmlOrigins()
```

Comment:
Updated to use existing function that handled this logic
```diff
- return _.keys(await Cypress.backend('get:rendered:html:origins')).concat([currentOrigin])
+ return await this.getAllHtmlOrigins()
```

Reply:
🙏